In [1]:
import scipy.stats as stats
import matplotlib.pyplot as plt
import numpy as np

Let’s say that there are two politicians who differ in their beliefs about whether the public is in favor an extra tax to support the national parks. Senator Smith thinks that only 40% of people are in favor of the tax, whereas Senator Jones thinks that 60% of people are in favor. They arrange to have a poll done to test this, which asks 1000 randomly selected people whether they support such a tax. The results are that 490 of the people in the polled sample were in favor of the tax. 

In [2]:
smith_p = 0.4
jones_p = 0.6
k = 490
n = 1000
p_data_given_smith = stats.binom.pmf(k=k, n=n, p=smith_p)
p_data_given_jones = stats.binom.pmf(k=k, n=n, p=jones_p)

In [3]:
bayes_factor = p_data_given_smith/p_data_given_jones
bayes_factor

3325.256730079536

The resulting Bayes Factor (3325.26) provides a measure of the evidence that the data provides regarding the two hypotheses - in this case, it tells us the data support Senator Smith more than 3000 times more strongly than they support Senator Jones.